### PR TITLE
Fix cron icon

### LINKED ISSUE #3849: Cron icon missing from new canvas
When you select a cron trigger, the clock icon is missing. It works fine for webhook and kafka, but not cron

### PR DESCRIPTION
**Pull request by taylordowns2000**
## Description

This PR fixes #3849

When switching a trigger to type cron in the collaborative editor, the icon disappeared due to missing cron_expression field in the trigger node data.

###  Solution

- Updated from-workflow.tsx to include type-specific fields (cron_expression, has_auth_method) when transforming trigger data
- Added KafkaTrigger type to workflow-diagram types for completeness
- Improved Trigger.tsx error handling to always display icon with fallback labels
   ("Not configured", "Invalid cron expression")

###  Testing

Added 14 unit tests covering all trigger types (webhook, cron, kafka) and edge cases (invalid/empty expressions). All 413 tests pass.

## Validation steps

1. Change the trigger type to cron on the experimental canvas and see the icon work properly!

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [x] Code generation (copilot but not intellisense)
- [x] Learning or fact checking
- [ ] Strategy / design
- [ ] Optimisation / refactoring
- [ ] Translation / spellchecking / doc gen
- [ ] Other
- [ ] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

## Pre-submission checklist

- [x] I have performed a **self-review** of my code.
- [x] I have implemented and tested all related **authorization policies**. (e.g., `:owner`, `:admin`, `:editor`, `:viewer`)
- [x] I have updated the **changelog**.
- [x] I have ticked a box in "AI usage" in this PR

### CODE REVIEW

File: assets/js/workflow-diagram/nodes/Trigger.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import { memo } from 'react';
-import { Position } from '@xyflow/react';
-import { ClockIcon, GlobeAltIcon } from '@heroicons/react/24/outline';
-import { kafkaIcon, lockClosedIcon } from '../components/trigger-icons';
-import cronstrue from 'cronstrue';
+import { memo } from "react";
```
Comment (josephjclark) [inline assets/js/workflow-diagram/nodes/Trigger.tsx]: Hey couldn't help notice these formatting diffs

Are we all on the same page with formatters? What's controlling that these days? I've had some precommit stuff running recently which has taken a lot of control away from me, but it should be handling this stuff right?

Reviewing all these PRs gets a lot harder when the diffs are so messy. And for what it's worth I'd prefer singlequotes

Comment (taylordowns2000) [reply assets/js/workflow-diagram/nodes/Trigger.tsx]: eish, that's terrible. sorry. let me double check the formatters - maybe related to the new git hooks or maybe related to my local vsCode setup. yes, single quotes the way to go (and i thought this was built in somewhere)

Comment (taylordowns2000) [reply assets/js/workflow-diagram/nodes/Trigger.tsx]: ah! @stuartc , was this intentional?
image
